### PR TITLE
Feat/coverage addresses and balance tests

### PR DESCRIPTION
**Pull request by kwek20**
Added a small amount of tests adding ~ 0.4% to coverage :/

### CODE REVIEW

File: sdk/tests/wallet/balance.rs [hunk 1]
```diff
@@ -165,6 +165,77 @@ async fn balance_expiration() -> Result<()> {
...
+        .retry_transaction_until_included(&tx.transaction_id, None, None)
+        .await?;
+    account_1.sync(None).await?;
+
+    // Balance should have transferred entirely
+    let balance_1: Balance = account_1.addresses_balance(vec![acc_1_addr.address()]).await?;
+    let balance_1_sync = account_1.balance().await?;
+    assert!(balance_1.base_coin().available() > 0);
+    assert_eq!(balance_1, balance_1_sync);
+    println!("{:?}", &balance_1_sync);
```
Comment (Thoralf-M) [inline sdk/tests/wallet/balance.rs]: Remove?

File: sdk/tests/client/addresses.rs [hunk 2]
```diff
@@ -40,6 +40,21 @@ async fn addresses() {
...
+
+    let opts = GetAddressesOptions::default()
+        .with_bech32_hrp(IOTA_TESTNET_BECH32_HRP)
+        .with_coin_type(IOTA_COIN_TYPE)
+        .with_range(0..1);
+    let public = secret_manager.generate_evm_addresses(opts.clone()).await.unwrap();
+    let internal = secret_manager.generate_evm_addresses(opts.internal()).await.unwrap();
+
+    assert_eq!(public[0], "0xb23e784f0464a30d536c961e414925eab6b3107d");
+    assert_eq!(internal[0], "0x98d8833ec4b82587d66207eb9c578fd0134c51b6");
```
Comment (thibault-martinez) [inline sdk/tests/client/addresses.rs:58]: Where did you get these from?

Comment (kwek20) [reply sdk/tests/client/addresses.rs:58]: from generating them :-)

Comment (kwek20) [reply sdk/tests/client/addresses.rs:58]: Do you want me to add a comment like yours:
```
// Address generated with bip32 path: [??, 4218, 0, 0, 0].
// This address was generated with a MnemonicSecretManager.
// Seed: 0x256a818b2aac458941f7274985a410e57fb750f3a3a67969ece5bd9ae7eef5b2.
```

Comment (thibault-martinez) [reply sdk/tests/client/addresses.rs:58]: Yeah but I mean, we should generate them with something else, otherwise the test is pointless

Comment (kwek20) [reply sdk/tests/client/addresses.rs:58]: not really, now its tested to be consistent in the future. What do you want me to generate it with? And whats the first number on BIP? Edit: 44 always, got it

Comment (thibault-martinez) [reply sdk/tests/client/addresses.rs:58]: Yeah but we shouldn't test it against itself. We should get the static values from another source to confirm it's correct.

Comment (Thoralf-M) [reply sdk/tests/client/addresses.rs:58]: Just tried with https://iancoleman.io/bip39/ and it works with the seed in BIP39 Seed and `m/44'/4218'/0'/0` in BIP32 Derivation Path
image

Comment (thibault-martinez) [reply sdk/tests/client/addresses.rs:58]: The comments I added in my PR are to explain I generated the values from a second source

Comment (thibault-martinez) [reply sdk/tests/client/addresses.rs:58]: Thanks @Thoralf-M , then Brord maybe adapt the comment to say it's been tested against that

Comment (kwek20) [reply sdk/tests/client/addresses.rs:58]: Nice tool, added it to the list ;) will update

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/tests/client/addresses.rs [hunk 3]
```diff
@@ -259,5 +278,39 @@ async fn search_address() -> Result<()> {
...
+    // not in range
+    let res =
+        iota_sdk::client::api::search_address(&secret_manager, IOTA_BECH32_HRP, IOTA_COIN_TYPE, 0, 0..9, &addresses[0])
+            .await;
+
+    if let Err(iota_sdk::client::Error::InputAddressNotFound { .. }) = res {
+        return Ok(());
+    }
+
+    panic!("should not have found search address range & public")
```
Comment (thibault-martinez) [inline sdk/tests/client/addresses.rs]: This is a bit weird tbh, can you just assert that res is the err please?

Comment (kwek20) [reply sdk/tests/client/addresses.rs]: Cant,. Error isnt PartialEq. ill put it in an else statement to make it slightly better

Comment (thibault-martinez) [reply sdk/tests/client/addresses.rs]: Don't think you need PartialEq if you use `matches!`

File: sdk/tests/client/node_api/core.rs [hunk 4]
```diff
@@ -266,9 +266,10 @@ async fn test_get_utxo_changes_by_index() {
     let node_info = client.get_info().await.unwrap();
 
     let r = client
-        .get_utxo_changes_by_index(node_info.node_info.status.latest_milestone.index)
+        .get_utxo_changes_by_index(node_info.node_info.status.confirmed_milestone.index)
         .await
         .unwrap();
+    assert_eq!(r.index, node_info.node_info.status.confirmed_milestone.index);
```
Comment (thibault-martinez) [inline sdk/tests/client/node_api/core.rs:272]: Add it to the other test as well?

File: sdk/tests/wallet/balance.rs [hunk 5]
```diff
@@ -165,6 +165,76 @@ async fn balance_expiration() -> Result<()> {
...
+        .retry_transaction_until_included(&tx.transaction_id, None, None)
+        .await?;
+    let balance_1_sync = account_1.sync(None).await?;
+
+    // Check the new address
+    let balance_1: Balance = account_1.addresses_balance(vec![acc_1_addr_2.address()]).await?;
+    assert_eq!(to_send / 2, balance_1.base_coin().available());
+
+    // Check old and new together
+    let balance_1_total: Balance = account_1
```
Comment (thibault-martinez) [inline sdk/tests/wallet/balance.rs]: Just curious, why explicit the type?

Comment (kwek20) [reply sdk/tests/wallet/balance.rs]: argh, my ide keeps adding that when i ctrl+click on the type preview.

File: sdk/tests/client/node_api/core.rs (hunk 4 above)
Comment (kwek20) [reply sdk/tests/client/node_api/core.rs:272]: it has an id, but the reposnse wont have an id, i suppose we can still test it but felt kinda useless

**Review by DaughterOfMars — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**